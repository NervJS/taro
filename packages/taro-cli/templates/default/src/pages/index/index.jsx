<%if (['react', 'preact'].includes(framework)) {-%>
import { Component<% if (typescript) {%>, PropsWithChildren<%}%> } from 'react'
<%}-%>
import { View, Text } from '@tarojs/components'
import './index.<%= cssExt %>'

export default class <%= _.capitalize(pageName) %> extends <% if (typescript) {%>Component<PropsWithChildren><%} else {%>Component<%}%> {

  componentWillMount () { }

  componentDidMount () { }

  componentWillUnmount () { }

  componentDidShow () { }

  componentDidHide () { }

  render () {
    return (
      <View className='<%= pageName %>'>
        <Text>Hello world!</Text>
      </View>
    )
  }
}
