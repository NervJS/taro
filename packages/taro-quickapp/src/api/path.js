// copy from https://github.com/jinder/path/blob/master/path.js
export function resolve () {
  var resolvedPath = ''
  var resolvedAbsolute = false

  for (var i = arguments.length - 1; i >= -1 && !resolvedAbsolute; i--) {
    var path = (i >= 0) ? arguments[i] : '/'

    if (!path) {
      continue
    }

    resolvedPath = path + '/' + resolvedPath
    resolvedAbsolute = path[0] === '/'
  }

  // At this point the path should be resolved to a full absolute path, but
  // handle relative paths to be safe (might happen when process.cwd() fails)

  // Normalize the path
  resolvedPath = normalizeArray(resolvedPath.split('/'),
    !resolvedAbsolute).join('/')

  return ((resolvedAbsolute ? '/' : '') + resolvedPath) || '.'
}

function normalizeArray (parts, allowAboveRoot) {
  var res = []
  for (var i = 0; i < parts.length; i++) {
    var p = parts[i]

    // ignore empty parts
    if (!p || p === '.') { continue }

    if (p === '..') {
      if (res.length && res[res.length - 1] !== '..') {
        res.pop()
      } else if (allowAboveRoot) {
        res.push('..')
      }
    } else {
      res.push(p)
    }
  }

  return res
}
