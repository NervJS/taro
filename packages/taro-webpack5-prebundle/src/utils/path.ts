export const addLeadingSlash = (url = '') => (url.charAt(0) === '/' ? url : '/' + url)
export const addTrailingSlash = (url = '') => (url.charAt(url.length - 1) === '/' ? url : url + '/')
export function parsePublicPath (publicPath = '/') {
  return ['', 'auto'].includes(publicPath) ? publicPath :  addTrailingSlash(publicPath)
}