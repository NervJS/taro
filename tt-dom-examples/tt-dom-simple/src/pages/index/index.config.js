// eslint-disable-next-line no-undef
export default definePageConfig({
  navigationBarTitleText: '首页'
})
