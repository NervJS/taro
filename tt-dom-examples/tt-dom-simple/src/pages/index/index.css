.grid-container {
    display: flex;
    flex-wrap: wrap;
    width: 100vw;
  }
  
.grid-box {
    width: 60px;
    height: 60px;
    margin: 3px;
    padding: 0;
}

.font {
    font-size: 24px;
}

.controls {
    text-align: center;
}


.button {
    display: inline-block;
    width: 150px;
    padding: 10px 20px;
    font-size: 16px;
    cursor: pointer;
    border: none;
    border-radius: 5px;
    margin-left: 20px;
    margin-bottom: 10px;
    color: white;
    transition: background-color 0.3s ease;
}



.start {
    background-color: #28a745; /* 绿色 */
}

.start:hover {
    background-color: #218838;
}

.clear {
    background-color: #dc3545; /* 红色 */
}

.clear:hover {
    background-color: #c82333;
}