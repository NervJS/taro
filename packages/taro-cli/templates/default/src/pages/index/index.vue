<template>
  <view class="<%= pageName %>">
    <text>{{ msg }}</text>
  </view>
</template>

<script>
<%if (framework === 'vue3') {-%>
import { ref } from 'vue'
<%}-%>
import './index.<%= cssExt %>'

export default {
<%if (framework === 'vue') {-%>
  data () {
    return {
      msg: 'Hello world!'
    }
  }
<%}-%>
<%if (framework === 'vue3') {-%>
  setup () {
    const msg = ref('Hello world')
    return {
      msg
    }
  }
<%}-%>
}
</script>
