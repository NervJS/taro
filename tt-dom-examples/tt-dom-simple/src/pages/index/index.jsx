import './index.css'

import { View } from '@tarojs/components'
import { useEffect, useRef, useState } from 'react'

const TOTAL = 1000
let timer = -1

const GridPage = () => {
  const [colors, setColors] = useState(
    () => new Array(TOTAL).fill(false) // false = 灰色，true = 蓝色
  )
  const currentIndex = useRef(0)

  useEffect(() => {
    console.log(' useEffect ')
  }, [])

  function start() {
    if (timer > 0) return
    timer = setInterval(() => {
      if (currentIndex.current >= TOTAL) {
        clearInterval(timer)
        timer = -1
        currentIndex.current = 0
        return
      }

      setColors((prev) => {
        const updated = [...prev]
        updated[currentIndex.current] = true
        currentIndex.current += 1
        return updated
      })
    }, 20)
  }

  function clear() {
    setColors(() => new Array(TOTAL).fill(false))
  }
  return (
    <View>
      <View className="controls">
        <View className="button start" onClick={start}>开始</View>
        <View className="button clear" onClick={clear}>清除</View>
      </View>
      <View className="grid-container">
        {colors
          .map((isActive, index) => (
            <View
              key={index}
              className="grid-box"
            >
              <View>
                <View>
                  <View>
                    <View style={{ backgroundColor: isActive ? '#3498db' : '#eee' }}>
                      <View className="font">{index}</View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          ))
        }
      </View>
    </View>

  )
}

export default GridPage
