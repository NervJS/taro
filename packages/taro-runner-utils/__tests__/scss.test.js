import path from 'path'
import { getBundleResult, getBundleContent, getSassLoaderOption } from '../dist/index'

describe('getBundleResult', () => {
  const filePath = path.resolve(__dirname, '.', 'styles/variables.scss')
  const fileConfig = {
    name: 'styles/variables.scss',
    path: path.resolve(__dirname, '.')
  }

  test('test file path', async () => {
    const res = await getBundleResult(filePath)
    expect(res.bundledContent).toMatchSnapshot()
  })

  test('test file name and project directory path', async () => {
    const res = await getBundleResult(fileConfig.name, fileConfig.path)
    expect(res.bundledContent).toMatchSnapshot()
  })
})

describe('getBundleContent', () => {
  const sassResourcePath = {
    resource: path.resolve(__dirname, '.', 'styles/variables.scss')
  }
  const sassResourcePathWithDirectory = {
    resource: 'styles/variables.scss',
    projectDirectory: path.resolve(__dirname, '.')
  }
  const sassResourceArray = {
    resource: [
      path.resolve(__dirname, '.', 'styles/variables.scss'),
      path.resolve(__dirname, '.', 'styles/mixins.scss')
    ]
  }
  const sassResourceArrayWithDirectory = {
    resource: [
      'styles/variables.scss',
      'styles/mixins.scss'
    ],
    projectDirectory: path.resolve(__dirname, '.')
  }

  test('test sass resource path', async () => {
    const data = sassResourcePath
    const res = await getBundleContent(data.resource)
    expect(res).toMatchSnapshot()
  })
  test('test sass resource path with directory', async () => {
    const data = sassResourcePathWithDirectory
    const res = await getBundleContent(data.resource, data.projectDirectory)
    expect(res).toMatchSnapshot()
  })

  test('test sass resource array', async () => {
    const data = sassResourceArray
    const res = await getBundleContent(data.resource)
    expect(res).toMatchSnapshot()
  })
  test('test sass resource array with directory', async () => {
    const data = sassResourceArrayWithDirectory
    const res = await getBundleContent(data.resource, data.projectDirectory)
    expect(res).toMatchSnapshot()
  })
})

describe('getSassLoaderOption', () => {
  const buildConfig = {
    publicPath: '/',
    staticDirectory: 'static',
    deviceRatio: { 640: 1.17, 750: 1, 828: 0.905 },
    designWidth: 750,
    outputRoot: 'dist',
    sass: {
      resource: [
        'styles/variables.scss',
        'styles/mixins.scss'
      ],
      projectDirectory: path.resolve(__dirname, '.')
    },
    sassLoaderOption: {
      implementation: 'dart-sass'
    },
    sourceRoot: 'src'
  }

  test('test get sass loader option', async () => {
    const res = await getSassLoaderOption(buildConfig)
    expect(res).toMatchSnapshot()
  })
})
