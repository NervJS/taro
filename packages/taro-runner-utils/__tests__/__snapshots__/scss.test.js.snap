// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBundleContent test sass resource array 1`] = `
"$color-ui-info: #78A4FA !default;
$color-ui-positive: #13CE66 !default;
$color-ui-negative: #FF4949 !default;
$color-ui-warning: #FFC82C !default;
@mixin transform($property) {
  -webkit-transform: $property;
  -ms-transform: $property;
  transform: $property;
}
"
`;

exports[`getBundleContent test sass resource array with directory 1`] = `
"$color-ui-info: #78A4FA !default;
$color-ui-positive: #13CE66 !default;
$color-ui-negative: #FF4949 !default;
$color-ui-warning: #FFC82C !default;
@mixin transform($property) {
  -webkit-transform: $property;
  -ms-transform: $property;
  transform: $property;
}
"
`;

exports[`getBundleContent test sass resource path 1`] = `
"$color-ui-info: #78A4FA !default;
$color-ui-positive: #13CE66 !default;
$color-ui-negative: #FF4949 !default;
$color-ui-warning: #FFC82C !default;
"
`;

exports[`getBundleContent test sass resource path with directory 1`] = `
"$color-ui-info: #78A4FA !default;
$color-ui-positive: #13CE66 !default;
$color-ui-negative: #FF4949 !default;
$color-ui-warning: #FFC82C !default;
"
`;

exports[`getBundleResult test file name and project directory path 1`] = `
"$color-ui-info: #78A4FA !default;
$color-ui-positive: #13CE66 !default;
$color-ui-negative: #FF4949 !default;
$color-ui-warning: #FFC82C !default;
"
`;

exports[`getBundleResult test file path 1`] = `
"$color-ui-info: #78A4FA !default;
$color-ui-positive: #13CE66 !default;
$color-ui-negative: #FF4949 !default;
$color-ui-warning: #FFC82C !default;
"
`;

exports[`getSassLoaderOption test get sass loader option 1`] = `
Object {
  "data": "$color-ui-info: #78A4FA !default;
$color-ui-positive: #13CE66 !default;
$color-ui-negative: #FF4949 !default;
$color-ui-warning: #FFC82C !default;
@mixin transform($property) {
  -webkit-transform: $property;
  -ms-transform: $property;
  transform: $property;
}
",
  "implementation": "dart-sass",
}
`;
